src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -46,29 +46,45 @@ struct DistributedSearchTreeImpl
 {
   // spatial queries
   template <typename DistributedTree, typename ExecutionSpace,
-            typename Predicates, typename Indices, typename Offset,
-            typename Ranks>
-  static std::enable_if_t<Kokkos::is_view<Indices>{} &&
-                          Kokkos::is_view<Offset>{} && Kokkos::is_view<Ranks>{}>
+            typename Predicates, typename IndicesAndRanks, typename Offset>
+  static std::enable_if_t<Kokkos::is_view<IndicesAndRanks>{} &&
+                          Kokkos::is_view<Offset>{}>
   queryDispatch(SpatialPredicateTag, DistributedTree const &tree,
                 ExecutionSpace const &space, Predicates const &queries,
-                Indices &indices, Offset &offset, Ranks &ranks)
+                IndicesAndRanks &values, Offset &offset)
   {
-    Kokkos::View<Kokkos::pair<int, int> *, DeviceType> out("pairs_index_rank",
-                                                           0);
     int comm_rank;
     MPI_Comm_rank(tree._comm, &comm_rank);
     queryDispatch(SpatialPredicateTag{}, tree, space, queries,
-                  CallbackDefaultSpatialPredicateWithRank{comm_rank}, out,
+                  CallbackDefaultSpatialPredicateWithRank{comm_rank}, values,
                   offset);
+  }
+
+  template <typename DistributedTree, typename ExecutionSpace,
+            typename Predicates, typename Indices, typename Offset,
+            typename Ranks>
+  /*deprecated*/ static std::enable_if_t<Kokkos::is_view<Indices>{} &&
+                                         Kokkos::is_view<Offset>{} &&
+                                         Kokkos::is_view<Ranks>{}>
+  queryDispatch(SpatialPredicateTag, DistributedTree const &tree,
+                ExecutionSpace const &space, Predicates const &queries,
+                Indices &indices, Offset &offset, Ranks &ranks)
+  {
+    struct PairIndexRank
+    {
+      int index;
+      int rank;
+    };
```

Comment:
NVCC does not like that :)

Reply:
It doesn't like when you stick it in inappropriate places :)

Reply:
Speaking of inappropriate 🙄